### PR TITLE
feat: your picks section, recently view section, dynamic search title

### PR DESCRIPTION
**Pull request by loks0n**
## What does this PR do?

- Removes some inline styles in favour of Pink design utility classes.
- Adds a dynamic title to the search results page, based on the search query.
- Adds a "Your Picks" section to the homepage, containing the users last 3 upvoted projects. User must have an account for this to display.
- Adds a "Recently Viewed" section to the homepage, containing the users last three viewed projects.

## Test Plan

Manual testing

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/index.tsx [hunk 1]
```diff
@@ -82,9 +90,35 @@ export const head: DocumentHead = () => ({
...
   const homeDataSignal = useHomeData();
   const homeData = homeDataSignal.value;
 
+  const yourPicks = useSignal<Project[] | undefined>([]);
+  useVisibleTask$(async () => {
+    if (!account.value || upvotes.value.length < 3) return;
+    yourPicks.value = await Promise.all(
+      upvotes.value
+        .slice(0, 3)
+        .map(async (upvote) => AppwriteService.getProject(upvote.projectId))
```
Comment (Meldiron) [inline src/routes/index.tsx:104]: Would it be possible to do `getProjects` and provide all projectsIds as array?
in `listDocuments` we would do `Query.equals('$id', [id1, id2i ,id3, id4..]`..

While yours performs well on frontend thanks to Promise.all, doing only 1 HTTP request can perform good also on server-side

**Review by Meldiron — COMMENTED** [review]
Love the PR, left some comments

File: src/routes/index.tsx (hunk 1 above)
Comment (loks0n) [reply src/routes/index.tsx]: Done

File: src/routes/index.tsx [hunk 2]
```diff
@@ -82,9 +90,35 @@ export const head: DocumentHead = () => ({
...
+  useVisibleTask$(async () => {
+    const visitedProjects = JSON.parse(
+      localStorage.getItem("visitedProjects") ?? "[]"
+    ) as string[];
+    if (visitedProjects.length < 3) return;
+
+    recentlyVisited.value = await Promise.all(
+      visitedProjects
+        .slice(0, 3)
+        .map(async (id) => AppwriteService.getProject(id))
```
[the remark this answers is not in the document]
Comment (loks0n) [reply src/routes/index.tsx]: Done

**Review by Meldiron — APPROVED** [review]

File: src/routes/search/index.tsx [hunk 3]
```diff
@@ -64,6 +64,36 @@ const generateQueries = (url: URL) => {
...
+
+  if (useCase) {
+    return `${useCase} projects`;
+  }
+
+  if (service) {
+    return `Using ${service}`;
+  }
+
+  return "Search";
```
Comment (christyjacob4) [inline src/routes/search/index.tsx:94]:
```suggestion
  return "Search Results";
```

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**